multinet/auth/decorators.py

```diff
@@ -44,19 +43,40 @@ def require_permission_inner(func: Any) -> Any:
         @wraps(func)
         def wrapper(*args, **kwargs) -> Any:
             workspace, user = _get_workspace_and_user(*args, **kwargs)
-            user_perm = workspace.get_user_permission(user)
-            if workspace.public and minimum_permission == WorkspacePermission.reader:
+            user_permission: WorkspaceRole = workspace.get_user_permission(user)
+
+            if (
+                workspace.public and minimum_permission == WorkspaceRoleChoice.READER
+            ) or workspace.owner == user:
                 return func(*args, **kwargs)
 
-            if user_perm is None:
+            if user_permission is None:
                 if workspace.public:
                     return HttpResponseForbidden()
                 return HttpResponseNotFound()
 
-            if user_perm.value >= minimum_permission.value:
+            if user_permission.role >= minimum_permission:
                 return func(*args, **kwargs)
             return HttpResponseForbidden()
```

Comment:
I think with this suggestion if the workspace was public and the user had no permission, they'd get a 404 instead of being able to use the endpoint. In this scenario the user is implicitly a reader. This can be solved by first checking the conditions that allow use of the endpoint, and THEN determining if the correct response is a 403 or 404. So I've basically adopted this suggestion, but the order is different.

Reply:
Good point, I missed that.